framework/include/mesh/FaceInfo.h

```diff
@@ -36,7 +37,14 @@ class Node;
 class FaceInfo
 {
 public:
-  FaceInfo(const ElemInfo * const elem_info, const unsigned int side, const dof_id_type id);
+  // Simple constructor for unit tests
+  FaceInfo(const ElemInfo * elem_info, unsigned int side, const dof_id_type id);
+  // Optimized constructor for the MooseMesh to use
+  FaceInfo(const ElemInfo * elem_info,
+           unsigned int side,
+           const dof_id_type id,
+           libMesh::ElemSideBuilder & side_builder,
+           libMesh::FEBase * fe);
```

Comment:
```suggestion
           libMesh::FEBase & fe);
```

There's really no need to pass this as a pointer as it should never be null